### PR TITLE
Update to SeqAn3 implementation

### PR DESCRIPTION
**Pull request by eseiler**
* Update to use SeqAn3 implementation
* Offer same functionality as IBF app

* [X] build
* [x] search

* Precomputed + parts do not work together (parts has no effect). Do we need this already now?

### CODE REVIEW

File: src/build_min_ibf.cpp [hunk 1]
```diff
@@ -213,18 +350,129 @@ int main(int argc, char ** argv)
...
+
+    // ==========================================
+    // Run correct method.
+    // ==========================================
+    if (arguments.binary)
+    {
+        compute_minimisers(arguments);
+        return 0;
+    }
+    if (arguments.precomputed)
```
Comment (MitraDarja) [inline src/build_min_ibf.cpp:465]: Shouldn't this be an else if?

File: src/build_min_ibf.cpp [hunk 2]
```diff
@@ -213,18 +350,129 @@ int main(int argc, char ** argv)
...
+    {
+        if (arguments.compressed)
+            build_from_binary<true>(arguments);
+        else
+            build_from_binary<false>(arguments);
+        return 0;
+    }
+    if (arguments.compressed)
```
Comment (MitraDarja) [inline src/build_min_ibf.cpp:473]: And here as well, you don't want to build the IBF twice...

**Comment by MitraDarja** [comment]
@eseiler 
When I use this branch to build an ibf with precomputed minimiser files, then I run into a "Not enough bins."-Error, which seems to make sense to me because the range that is given to the technical_binnig directory is empty? Or did I understand the code incorrectly?

For the paper, I do not think, we need precomputed and parts to work together, because I was not planing on using them together.

**Comment by eseiler** [comment]
> @eseiler
> When I use this branch to build an ibf with precomputed minimiser files, then I run into a "Not enough bins."-Error, which seems to make sense to me because the range that is given to the technical_binnig directory is empty? Or did I understand the code incorrectly?

Can you change the constructor in `build_from_binary` from

```cpp
    seqan3::technical_binning_directory ibf{std::vector<std::vector<seqan3::dna4>>{},
                                            seqan3::views::minimiser_hash(seqan3::ungapped{arguments.k},
                                                                          seqan3::window_size{arguments.w}),
                                            cfg};
```

to

```
    seqan3::technical_binning_directory ibf{std::vector<std::vector<seqan3::dna4>>{},
                                            seqan3::views::minimiser_hash(seqan3::ungapped{arguments.k},
                                                                          seqan3::window_size{arguments.w}),
                                            cfg,
                                            true};
```

and try again?

**Comment by MitraDarja** [comment]
@eseiler 

Then I get no error message, but I also don't get an output.

**Comment by eseiler** [comment]
superseded by https://github.com/seqan/raptor

**Pull request closed without merging**